templates/connect-inject-mutatingwebhook.yaml

```diff
@@ -12,6 +16,7 @@ metadata:
     release: {{ .Release.Name }}
 webhooks:
   - name: {{ template "consul.fullname" . }}-connect-injector.consul.hashicorp.com
+    failurePolicy: Fail
```

Comment:
To be compliant with the `admissionregistration.k8s.io/v1` spec of webhooks, a few additional changes need to be made to the webhooks. We need to add the `sideEffects: None` to the webhook here along with `failurePolicy`.

Additionally, every webhook in this file and `templates/controller-mutatingwebhookconfiguration.yaml` need to add an `admissionReviewVersions` field to each of their webhooks as well. I was able to add those and get those webhooks to get submitted successfully. 
```
  admissionReviewVersions:
  - "v1beta1"
  - "v1"
```

Reply:
Sorry for missing these out, I have added them.